src/server/pfs/s3/object.go

```diff
@@ -43,14 +42,36 @@ func (c *controller) GetObject(r *http.Request, bucketName, file, version string
 		commitID = version
 	}
 
-	fileInfo, err := pc.InspectFile(bucket.Commit, file)
+	// We use listFileResult[0] rather than InspectFile result since InspectFile
+	// on a path that has both a file and a directory in it returns the
+	// directory. However, ListFile will show it as a file, if it exists.
+	var firstFile *pfs.FileInfo
+	err = pc.ListFile(bucket.Commit, file, func(fi *pfs.FileInfo) (retErr error) {
+		if firstFile == nil {
+			firstFile = fi
+		}
+		// TODO: would be nice to stop iterating after the first one
+		return nil
+	})
 	if err != nil {
 		return nil, maybeNotFoundError(r, err)
 	}
+	if firstFile == nil {
+		// we never set it, probably zero results
+		return nil, s2.NoSuchKeyError(r)
+	}
+	fileInfo := firstFile
+
+	// the exact object named does not exist, but perhaps is a "directory".
+	// "directories" do not actually exist, and certainly cannot be read.
+	// ("seeker can't seek")
+	if fileInfo.File.Path[1:] != file {
```

Comment:
hmm why is it `[1:]`? I would expect since we trim the suffix, it would be something like `[:len(file)-1]`?

Reply:
what do you think about moving this check up to the `pc.ListFile` iteration's call back function?
Something like
```Go
if fi.File.Path[1:] == file {
    firstFile  = fi
}
```

Reply:
It's `[1:]` because Path starts with a `/` but file doesn't